utils/python/CIME/st_archive.py

```diff
@@ -0,0 +1,190 @@
+        logger.info('component name = %s ' % comp)
+        logger.info('rootdir = %s' % rootdir)
+        logger.info('multiple-instance support = %s ' % multi)
+        casename = case.get_value('CASENAME')
+        dout_s_root = case.get_value('DOUT_S_ROOT')
+        for file_extension in archive.get_nodes('file_extension', root=archive_spec_node):
+            suffix = file_extension.attrib['regex_suffix']
+            subdir = archive.get_node('subdir', root=file_extension).text
+            keep_last_in_rundir = archive.get_node('keep_last_in_rundir', root=file_extension).text
+            logger.info('\n  ***** File extension specification')
+            logger.info('  regex_suffix = %s ' % suffix)
+            logger.info('  subdir = %s ' % join(dout_s_root, rootdir, subdir))
+            logger.info('  keep_last_in_rundir %s = ' % keep_last_in_rundir)
+
+
+def listArchive(dirname):
+    logger.debug('In listArchive: %s ...' % dirname)
+    logger.info('%s ' % dirname)
+    if isdir(dirname):
+        list_dirs = os.walk(dirname)
+        for root, dirs, files in list_dirs:
+            for f in files:
+                print join(root, f)
+
+
+def getDatename(case):
+    logger.debug('In getDatename...')
+    rundir = case.get_value('RUNDIR')
+    expect(isdir(rundir), 'Cannot open directory %s ' % rundir)
+    files = sorted(glob.glob(rundir + '/*cpl.r*.nc'))
+    rfile = files[-1]
+    names = rfile.split('.')
+    if names:
+        datename = names[-2]
+        logger.debug('Cpl dateName: %s ' % datename)
+        return datename
+    expect(False, 'Cannot find a cpl.r.*.nc file in directory %s ' % rundir)
+
+
+def moveFiles(case, keep_last_in_rundir, suffix, destdir, restdir, datename, runfiles):
+    logger.debug('In moveFiles...')
+    rundir = case.get_value('RUNDIR')
+    p = re.compile(suffix)
+    files = [ f for f in os.listdir(rundir) if p.match(f) ]
+    numfiles = len(files)
+    if numfiles > 0:
+        if keep_last_in_rundir:
+            keepfile = files[-1]
+            shutil.copy(join(rundir, keepfile), join(destdir, keepfile))
+            logger.debug('keepfile = %s ' % keepfile)
+            for filename in files:
+                if filename != keepfile and isfile(filename):
+                    shutil.move(join(rundir, filename), join(destdir, filename))
+
+        else:
+            for filename in files:
+                shutil.move(join(rundir, filename), join(destdir, filename))
+
+    for filename in files:
+        runfiles.remove(filename)
+
+    counter = 0
+    for filename in files:
+        if datename in filename:
+            counter = counter + 1
+            if counter > 1:
+                expect(False, 'Multiple restart files found for suffix %s ' % suffix)
+            restfile = basename(filename)
+            if isfile(join(destdir, restfile)):
+                shutil.copy(join(destdir, restfile), join(restdir, restfile))
+
+
+def archiveProcess(case, archive, datename, runfiles):
```

Comment:
I'd prefer "function_name_style" instead of functionNameStyle in order to be consistent with the rest of the codebase.